### PR TITLE
Add AWS ECS cluster detection

### PR DESCRIPTION
**Pull request by bjrara**
**Description:**

< Describe what is being changed or added.
  Ex. Bug fix  - Describe the bug and how this fixes it.
  Ex. Feature addition - Describe what this provides and why. >

**[aws.ecs.cluster.arn](https://opentelemetry.io/docs/specs/semconv/attributes-registry/aws/)** is not populated in the ecs resource attributes.

This PR reads `Cluster` from ECS task metadata API, and sets `aws.ecs.cluster.arn` into resource attributes.

**Testing:**
UT passed.

### CODE REVIEW

**Review by wangzlei — APPROVED** [review]

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java
Comment (srprash) [inline aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:247]: nb: should we do a more comprehensive check to be sure the `cluster` is in ARN format? Something like
```
cluster.startsWith("arn:aws:ecs:")
```

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/IncubatingAttributes.java
Comment (srprash) [inline aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/IncubatingAttributes.java:66]: Probably out of scope of this PR. But I don't see `aws.ecs.task.id` being populated while it is mentioned in the [spec](https://opentelemetry.io/docs/specs/semconv/attributes-registry/aws/#aws-ecs-attributes). Any plans on adding that too for being fully spec compliant?

**Review by srprash — APPROVED** [review]
LGTM. Minor comments.

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java
Comment (bjrara) [reply aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/EcsResource.java:247]: > Cluster name must be 1 to 255 characters. Valid characters are a-z, A-Z, 0-9, hyphens (-), and underscores (_).

Checking `:` and `arn:aws:ecs:` gives us the same result.

File: aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/IncubatingAttributes.java
Comment (bjrara) [reply aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/IncubatingAttributes.java:66]: `aws.ecs.task.id` is recently added, and that's probably why it isn't implemented.

Comment (bjrara) [reply aws-resources/src/main/java/io/opentelemetry/contrib/aws/resource/IncubatingAttributes.java:66]: We can do that in another PR, but I'll keep this clean and short.

**Comment by wangzlei** [comment]
@open-telemetry/java-contrib-approvers Please help review, thanks!

**Comment by wangzlei** [comment]
@trask @jkwatson Would you help take a look?

**Review by jkwatson — APPROVED** [review]
If @srprash approves, I'm fine merging.

**Comment by jkwatson** [comment]
> If @srprash approves, I'm fine merging.

But I'm not a maintainer over here, so someone who is will need to approve before we can merge.

**Pull request merged**
